src/main/java/com/cloudbees/hudson/plugins/folder/properties/FolderCredentialsProvider.java

```diff
@@ -178,7 +191,20 @@
                                                                    @NonNull CredentialsMatcher matcher) {
         ListBoxModel result = new ListBoxModel();
         Set<String> ids = new HashSet<>();
+
+        // Check if Authentication is authorised to list the credentials
+        Boolean isAuthorised = false;
         if (ACL.SYSTEM.equals(authentication)) {
+            isAuthorised = true;
+        }
+        if (itemGroup instanceof AbstractFolder) { 
+            final AbstractFolder<?> folder = AbstractFolder.class.cast(itemGroup);
+            if (folder.hasPermission(authentication, CredentialsProvider.USE_ITEM)) {
```

Comment:
I think this should allow for listing credentials if the user can also configure the itemgroup (as they may be able to select a credential that the system (not they themselves) will use later

(note the follow syntax has not been checked)

```suggestion
            if (folder.hasPermission(authentication, CredentialsProvider.USE_ITEM) || folder.hasPermission(authentication, Item.CONFIGURE)) {
```

@daniel-beck  I always have to double think this - but for listing the credential IDs it should be you have configure on the item, or the ability to use a credntials.  (configure so you can select a credential that the job/system can use even if you can not use it, USE_ITEM incase you can not configure the job (e.g. for the pipeline snippet generator at the job level and the pipeline is "as-code")?

Reply:
Hi, I'm assuming we're waiting for @daniel-beck 's response on this. Is there any indication on when we can expect that?